### PR TITLE
Refactor React class component into functional components

### PR DESCRIPTION
**Pull request by DavidMatalik**

### CODE REVIEW

File: src/components/CVSectionForm.js [hunk 1]
```diff
@@ -1,109 +1,87 @@
...
-          <CVSectionButton
-            value='Edit'
-            handleAction={this.handleEdit}
-          ></CVSectionButton>
-          <CVSectionButton
-            value='Submit'
-            handleAction={this.handleSubmit}
-          ></CVSectionButton>
+  // Or we want to have div's with latest input values
+  if (!edit) {
```
Comment (RichStone) [inline src/components/CVSectionForm.js:59]: this component smells like it's violating the SRP, but I'm  not sure now how we would go about fixing it :thinking:

**Review by RichStone — COMMENTED** [review]
LGTM

how do you like func components vs the old class components?

do you know how/why yarn.lock got updated and ended up in the PR?
(btw, I had a look on how to keep our package-lock.jsons / yarn.locks consistent, one thing we could try: https://stackoverflow.com/a/50798589/5925094)

**Review by RichStone — APPROVED** [review]
thanks for letting me have a look at this! :)

File: src/components/CVSectionForm.js (hunk 1 above)
Comment (DavidMatalik) [reply src/components/CVSectionForm.js:59]: Why do you think it's violating the SRP?

**Comment by DavidMatalik** [comment]
> do you know how/why yarn.lock got updated and ended up in the PR?

When I remember it right it was updated when I removed import statements from the first file. If I understand it right, then Babel makes JSX work for functional components. Maybe this has something to do with it?

**Comment by DavidMatalik** [comment]
> how do you like func components vs the old class components?

I like that the code is much shorter. BUT I found it clearer with the self explaining functions componentDidMount, componentWillUnmount and so on what exactly to put inside of them. And when to use them. I think it is very good approch to first learn about the class structure and lifecycle methods and then about functional components - easier to understand functional component this way. Thumbs up for TOP.

**Comment by RichStone** [comment]
> > do you know how/why yarn.lock got updated and ended up in the PR?
> 
> When I remember it right it was updated when I removed import statements from the first file. If I understand it right, then Babel makes JSX work for functional components. Maybe this has something to do with it?

Oh, no, I don't think this should impact something in the yarn.lock, as long as you don't touch dependencies the package.json..

File: src/components/CVSectionForm.js (hunk 1 above)
Comment (RichStone) [reply src/components/CVSectionForm.js:59]: it looks like the edit state needs a lot of special handling, as if it was its own component...

**Comment by RichStone** [comment]
are you always working from super clean git states? (i.e. doing `git status` a lot and seeing that only changed files are really changed)

we should keep an eye on the changing .locks :grimacing:

File: src/components/CVSectionForm.js (hunk 1 above)
Comment (DavidMatalik) [reply src/components/CVSectionForm.js:59]: Right... I'll put it into the creations-showcase in the Odin discord. And ask specifically about it. Let's see what inputs come from there. =)

**Review by DavidMatalik — COMMENTED** [review]
> are you always working from super clean git states? (i.e. doing `git status` a lot and seeing that only changed files are really changed)

Yes I am doing git status a lot, coming more and more into this routine. Maybe sometimes I forget about i or do a `git commit -am 'commit-message'`.

**Pull request merged**